rust/src/types/memory.rs

```diff
@@ -1,20 +1,33 @@
-use std::{cmp::max, iter};
+use std::{cmp::max, iter, sync::Mutex};
 
 use crate::{
     types::{u256, FailStatus},
     utils::{word_size, Gas},
 };
 
+static REUSABLE_MEMORY: Mutex<Option<Vec<u8>>> = Mutex::new(None);
```

Comment:
same question as with the stack. How does this work with the concurrent use in the CT? 

And how does this deal with recursion?

Reply:
If there is no allocation which can be reused because it is already in use, a new allocation is created. And when in Drop there is already one allocation in REUSABLE_STACK then one of those allocations (the one that was previously put there) will get deallocated. I will rewrite it to Mutex<Vec<Vec<u8>> and see how that performs...